runtime/doc/lsp.txt

```diff
@@ -800,6 +796,55 @@ set_log_level({level})                               *vim.lsp.set_log_level()*
                 See also: ~
                     |vim.lsp.log_levels|
 
+start({config}, {opts})                                      *vim.lsp.start()*
+                Create a new LSP client and start a language server or reuses
+                an already running client if one is found matching `name` and
+                `root_dir`. Attaches the current buffer to the client.
+
+                Example:
+>
+
+    vim.lsp.start({
+       name = 'my-server-name',
+       cmd = {'name-of-language-server-executable'},
+       root_dir = vim.fs.dirname(vim.fs.find('.git')[1]),
+    })
+<
+
+                See |lsp.start_client| for all available options. The most
+                important are:
+
+                `name` is an arbitrary name for the LSP client. It should be
+                unique per language server.
+
+                `cmd` the command as list - used to start the language server. The
+                command must be present in the `$PATH` environment variable or an absolute path to the executable.
+                Shell constructs like `~` are NOT expanded.
+
+                `root_dir` path to the project root. Language servers use this
+                information to discover metadata like the dependencies of your
+                project and they tend to index the contents within the project
+                folder. The example above uses |vim.fs.find| and
```

Comment:
I'd suggest explicitly mentioning that `root_dir` serves two purposes: beside the mentioned one, it is _also_ used by Neovim to keep track of whether to start a new server or attach to a running one. 

(This is implicit in the initial paragraph, but I think it wouldn't hurt to be as explicit as possible here, given the confusion about `root_dir` we've seen...)

Reply:
It's already mentioned in the first sentence, do you think it warrants repeating? I wasn't sure

Reply:
Yes, that's exactly what I think. Doesn't have to be a novel; just making clear that it's not _just_ for Language servers.

Reply:
On further thought, we should probably not further conflate the _server_ root dir and the _Neovim_ root dir (which is now handled explicitly by `reuse_client`). Maybe we should go even further in separating these concerns, but that shouldn't block this PR.